production/helm/loki/templates/_helpers.tpl

```diff
@@ -193,12 +193,39 @@ filesystem:
 Storage config for ruler
 */}}
 {{- define "loki.rulerStorageConfig" -}}
-{{- if or .Values.minio.enabled (eq .Values.loki.storage.type "s3") -}}
+{{- if .Values.minio.enabled -}}
 s3:
   bucketnames: {{ $.Values.loki.storage.bucketNames.ruler }}
+{{- else if eq .Values.loki.storage.type "s3" -}}
+{{- with .Values.loki.storage.s3 }}
+s3:
+  {{- with .s3 }}
+  s3: {{ . }}
```

Comment:
Is the indent right? Could you paste a rendered example?

Reply:
Proper indenting is done at [Value. yaml line 128](https://github.com/grafana/loki/blob/main/production/helm/loki/values.yaml#L128)
Here is generated config map:
```
auth_enabled: false
common:
  path_prefix: /var/loki
  storage:
    s3:
      bucketnames: <BUCKET_NAME>
      insecure: false
      region: us-east-1
      s3: s3://us-east-1/<BUCKET_NAME>
      s3forcepathstyle: false
limits_config:
  enforce_metric_name: false
  max_cache_freshness_per_query: 10m
  reject_old_samples: true
  reject_old_samples_max_age: 168h
  split_queries_by_interval: 15m
memberlist:
  join_members:
  - loki-memberlist
query_range:
  align_queries_with_step: true
ruler:
  storage:
    s3:
      bucketnames: <BUCKET_NAME>
      insecure: false
      region: us-east-1
      s3: s3://us-east-1/<BUCKET_NAME>
      s3forcepathstyle: false
```
Also ive deployed it to cluster, and now ruler.go properly authenticates to S3.